Assets/MixedRealityToolkit.Services/InputSystem/BaseNearInteractionGrabbable.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Base class for all NearInteractionGrabbables
+    ///
+    /// Technical details:
+    /// Provides a listing of near field grab proximity bounds.
+    /// This is used to detect if a contact point is near an object to turn on near field interactions
+    /// </summary>
+
+    public abstract class BaseNearInteractionGrabbable : MonoBehaviour
+    {
+        public static IReadOnlyCollection<BaseNearInteractionGrabbable> Instances => instances.AsReadOnly();
```

Comment:
This should use an overlap test similar to the one implemented in #4942. We're still trying to figure out how to make that work with UnityUI objects though.

Reply:
Thankfully we don't have to worry about UnityUI here since this is for grabbing objects. I'll take a look.